pkg/releaseutil/kind_sorter.go

```diff
@@ -67,47 +62,34 @@ var InstallOrder KindSortOrder = []string{
 	"APIService",
 }
 
+// addListKind adds the List equivalent for each resource kind in the list.
+func addListKind(resources KindSortOrder) KindSortOrder {
+	output := make(KindSortOrder, 0, 2*len(resources))
+	for _, resource := range resources {
+		output = append(output, resource)
+		output = append(output, resource+"List")
+	}
+	return output
+}
+
+// reverseKindOrder reverses the order of the KindSortOrder.
+func reverseKindOrder(resources KindSortOrder) KindSortOrder {
```

Comment:
fwiw, this would be `slices.Reverse(resources)` now (https://pkg.go.dev/slices#Reverse)

Reply:
In its current implementation the uninstall order is different in one aspect. The order of the uninstall of the Service resource. However, that seems to be incorrect. A statefulset has a dependency on a headless service.
The install order therefore creates the Service resource before the statefulset. It should however remove the statefulset before the dependent Service resource upon uninstall and doesn't do that.  This issue is now corrected as well by this PR.